zipline/data/bundles/csvdir.py

```diff
@@ -0,0 +1,215 @@
+                      asset_db_writer,
+                      minute_bar_writer,
+                      daily_bar_writer,
+                      adjustment_writer,
+                      calendar,
+                      start_session,
+                      end_session,
+                      cache,
+                      show_progress,
+                      output_dir,
+                      self.tframes,
+                      self.csvdir)
+
+
+@bundles.register("csvdir")
+def csvdir_bundle(environ,
+                  asset_db_writer,
+                  minute_bar_writer,
+                  daily_bar_writer,
+                  adjustment_writer,
+                  calendar,
+                  start_session,
+                  end_session,
+                  cache,
+                  show_progress,
+                  output_dir,
+                  tframes=None,
+                  csvdir=None):
+    """
+    Build a zipline data bundle from the directory with csv files.
+    """
+    if not csvdir:
+        csvdir = environ.get('CSVDIR')
+        if not csvdir:
+            raise ValueError("CSVDIR environment variable is not set")
+
+    if not os.path.isdir(csvdir):
+        raise ValueError("%s is not a directory" % csvdir)
+
+    if not tframes:
+        tframes = set(["daily", "minute"]).intersection(os.listdir(csvdir))
+
+        if not tframes:
+            raise ValueError("'daily' and 'minute' directories "
+                             "not found in '%s'" % csvdir)
+
+    divs_splits = {'divs': DataFrame(columns=['sid', 'amount',
+                                              'ex_date', 'record_date',
+                                              'declared_date', 'pay_date']),
+                   'splits': DataFrame(columns=['sid', 'ratio',
+                                                'effective_date'])}
+    for tframe in tframes:
+        ddir = os.path.join(csvdir, tframe)
+
+        symbols = sorted(item.split('.csv')[0]
+                         for item in os.listdir(ddir)
+                         if item.endswith('.csv'))
+        if not symbols:
+            raise ValueError("no <symbol>.csv files found in %s" % ddir)
+
+        dtype = [('start_date', 'datetime64[ns]'),
+                 ('end_date', 'datetime64[ns]'),
+                 ('auto_close_date', 'datetime64[ns]'),
+                 ('symbol', 'object')]
+        metadata = DataFrame(empty(len(symbols), dtype=dtype))
+
+        if tframe == 'minute':
+            writer = minute_bar_writer
+        else:
+            writer = daily_bar_writer
+
+        writer.write(_pricing_iter(ddir, symbols, metadata,
+                     divs_splits, show_progress),
+                     show_progress=show_progress)
+
+        # Hardcode the exchange to "CSVDIR" for all assets and (elsewhere)
+        # register "CSVDIR" to resolve to the NYSE calendar, because these
+        # are all equities and thus can use the NYSE calendar.
+        metadata['exchange'] = "CSVDIR"
```

Comment:
I implemented csvdir_equities function especially for this purpose. It can be used in extension.py to use another calendar or other custom parameters, e.g.
```
register('csvdir-custom', csvdir_equities(['daily', 'minute'], 
              "/Users/test/csvdir"),
             calendar_name='CUSTOM', minutes_per_day=24*60)
```

Reply:
Ah good call. That slipped past me.

Reply:
Code you share the code of your 'CUSTOM' calendar? :)